CMakeLists.txt

```diff
@@ -17,36 +17,41 @@ option(BUILD_QUSAT_TESTS "Also build tests for project")
 # build type settings
 set(default_build_type "Release")
 if (NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
-	message(STATUS "Setting build type to '${default_build_type}' as none was specified.")
-	set(CMAKE_BUILD_TYPE "${default_build_type}" CACHE STRING "Choose the type of build." FORCE)
-	set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release" "MinSizeRel" "RelWithDebInfo")
+    message(STATUS "Setting build type to '${default_build_type}' as none was specified.")
+    set(CMAKE_BUILD_TYPE "${default_build_type}" CACHE STRING "Choose the type of build." FORCE)
+    set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release" "MinSizeRel" "RelWithDebInfo")
 endif ()
 
 macro(check_submodule_present MODULENAME)
-	if(NOT EXISTS "${PROJECT_SOURCE_DIR}/extern/${MODULENAME}/CMakeLists.txt")
-		message(FATAL_ERROR "${MODULENAME} submodule not cloned properly. Please run `git submodule update --init --recursive` from the main project directory")
-	endif()
+    if (NOT EXISTS "${PROJECT_SOURCE_DIR}/extern/${MODULENAME}/CMakeLists.txt")
+        message(FATAL_ERROR "${MODULENAME} submodule not cloned properly. Please run `git submodule update --init --recursive` from the main project directory")
+    endif ()
 endmacro()
 
 check_submodule_present(qfr)
 
 # Add path for custom modules
-list (APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
 
 # search for Z3
 find_package(Z3 REQUIRED)
-if(Z3_FOUND AND NOT TARGET z3::z3lib)
-	add_library(z3::z3lib IMPORTED INTERFACE)
-	set_property(TARGET z3::z3lib PROPERTY INTERFACE_INCLUDE_DIRECTORIES ${Z3_CXX_INCLUDE_DIRS})
-	set_property(TARGET z3::z3lib PROPERTY INTERFACE_LINK_LIBRARIES ${Z3_LIBRARIES})
-endif()
+if (Z3_FOUND AND NOT TARGET z3::z3lib)
+    add_library(z3::z3lib IMPORTED INTERFACE)
+    set_property(TARGET z3::z3lib PROPERTY INTERFACE_INCLUDE_DIRECTORIES ${Z3_CXX_INCLUDE_DIRS})
+    set_property(TARGET z3::z3lib PROPERTY INTERFACE_LINK_LIBRARIES ${Z3_LIBRARIES})
+endif ()
 
 # add main library code
 add_subdirectory(src)
 
 # add test code
 if (BUILD_QUSAT_TESTS)
-	enable_testing()
-	include(GoogleTest)
-	add_subdirectory(test)
+    enable_testing()
+    include(GoogleTest)
+    add_subdirectory(test)
+endif ()
+
+# add Python binding code
+if (BINDINGS)
```

Comment:
you need to add the `BINDINGS` option to the configuration options above. Also it might make sense to directly also include the `DEPLOY` option for setting up the python wheels building.
```cmake
option(DEPLOY "Configure for deployment")
option(BINDINGS "Configure for building Python bindings")
```